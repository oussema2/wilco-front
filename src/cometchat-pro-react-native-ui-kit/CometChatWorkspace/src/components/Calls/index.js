export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
