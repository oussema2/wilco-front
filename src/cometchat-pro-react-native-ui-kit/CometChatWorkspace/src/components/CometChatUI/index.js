/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
