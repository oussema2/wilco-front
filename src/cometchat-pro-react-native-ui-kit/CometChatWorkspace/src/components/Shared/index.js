export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
