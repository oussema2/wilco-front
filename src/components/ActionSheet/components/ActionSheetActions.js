import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
