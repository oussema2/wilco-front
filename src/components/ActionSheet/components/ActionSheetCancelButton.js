import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
