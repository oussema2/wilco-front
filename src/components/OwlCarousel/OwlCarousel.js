import React, { useEffect, useRef, useState } from 'react';
import { Dimensions, View } from 'react-native';
import PropTypes from 'prop-types';
import Carousel from 'react-native-snap-carousel';
import { observer } from 'mobx-react-lite';
import { Pagination } from '../Pagination';

const OwlCarousel = ( {
	testID, items, renderItem, onWidthSet, setIndex, indexWhenCloseViewer
} ) => {
	const [ screenWidth ] = useState( Dimensions.get( 'window' ).width );
	const [ owlIndex, setOwlIndex ] = useState( 0 );
	const carouselRef = useRef( null );

	useEffect( () => {
		onWidthSet( screenWidth );
	}, [ ] );

	useEffect( () => {
		carouselRef.current.triggerRenderingHack();
	}, [ items ] );

	const _pagination = ( { items: itemsToPaginate } ) => (
		<Pagination length={itemsToPaginate.length} activeIndex={owlIndex} />
	);

	const _setCarouselIndex = ( _index ) => {
		setIndex( _index );
		setOwlIndex( _index );
	};

	return (
		<View testID={testID}>
			<Carousel
				ref={carouselRef}
				data={items}
				renderItem={renderItem}
				onSnapToItem={_setCarouselIndex}
				windowSize={3}
				sliderWidth={screenWidth}
				itemWidth={screenWidth}
				onScrollIndexChanged={_setCarouselIndex}
				firstItem={indexWhenCloseViewer}
			/>
			{ _pagination( { items } ) }
		</View>
	);
};

OwlCarousel.propTypes = {
	testID: PropTypes.string,
	items: PropTypes.arrayOf( PropTypes.any ),
	renderItem: PropTypes.func,
	onWidthSet: PropTypes.func,
	setIndex: PropTypes.func,
	indexWhenCloseViewer: PropTypes.number
};

OwlCarousel.defaultProps = {
	testID: '-Component',
	items: [],
	renderItem: () => {},
	onWidthSet: () => {},
	setIndex: () => {},
	indexWhenCloseViewer: 0
};

export default observer( OwlCarousel );
