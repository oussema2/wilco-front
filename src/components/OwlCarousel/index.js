export { default as OwlCarousel } from './OwlCarousel';
