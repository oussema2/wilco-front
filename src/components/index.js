export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
