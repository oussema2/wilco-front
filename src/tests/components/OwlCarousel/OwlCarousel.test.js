import React from 'react';
import { render } from '@testing-library/react-native';
import { OwlCarousel } from '../../../components/OwlCarousel';

describe( '<OwlCarousel />', () => {
	let component;
	const testID = 'testing-OwlCarousel-Component';
	const items = [ { id: 1 }, { id: 2 }, { id: 3 } ];
	const MOCKING_DIMENSION_WIDTH_VALUE = 1000;
	const MOCKING_DIMENSION_HEIGHT_VALUE = 750;

	const sources = [
		{ uri: 'https://example.photo' },
		{ uri: 'https://second.example.photo/2' }
	];

	const presenter = {
		imageSources: sources,
		imageViewerVisible: true,
		setImageViewerVisible: jest.fn()
	};

	const setUp = ( props ) => {
		component = render( <OwlCarousel
			testID={testID}
			items={items}
			{...props}
			postPresenter={presenter}
		/> );
	};

	const mockDimensions = ( { width, height } ) => {
		jest.doMock( 'react-native/Libraries/Utilities/Dimensions', () => ( {
			get: jest.fn().mockReturnValue( { width, height } )
		} ) );
	};

	jest.mock( 'react', () => ( {
		...jest.requireActual( 'react' ),
		useState: jest.fn()
	} ) );

	beforeEach( () => {
		mockDimensions( {
			width: MOCKING_DIMENSION_WIDTH_VALUE,
			height: MOCKING_DIMENSION_HEIGHT_VALUE
		} );
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the OwlCarousel component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with an onWidthSet callback', () => {
		it( 'calls the callback passing the screen width', () => {
			const onWidthSet = jest.fn();
			setUp( { onWidthSet } );

			expect( onWidthSet ).toHaveBeenCalledWith( MOCKING_DIMENSION_WIDTH_VALUE );
		} );
	} );
} );
