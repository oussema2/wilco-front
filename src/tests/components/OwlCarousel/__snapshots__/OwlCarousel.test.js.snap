// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OwlCarousel /> with the default props renders the OwlCarousel component correctly 1`] = `
<View
  testID="testing-OwlCarousel-Component"
>
  <RCTScrollView
    CellRendererComponent={[Function]}
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        Object {},
      ]
    }
    data={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
        Object {
          "id": 3,
        },
      ]
    }
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableScrollViewPanResponder={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={6}
    initialScrollIndex={0}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={1000}
    keyExtractor={[Function]}
    layout="default"
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollIndexChanged={[Function]}
    onSnapToItem={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    pointerEvents="auto"
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={1000}
    snapToOffsets={
      Array [
        0,
        1000,
        2000,
      ]
    }
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "width": 1000,
      }
    }
    useExperimentalSnap={false}
    useScrollView={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={3}
  >
    <View>
      <View
        CellRendererComponent={[Function]}
        cellKey="flatlist-item-0"
        fillRateHelper={
          FillRateHelper {
            "_anyBlankStartTime": null,
            "_enabled": false,
            "_getFrameMetrics": [Function],
            "_info": Info {
              "any_blank_count": 0,
              "any_blank_ms": 0,
              "any_blank_speed_sum": 0,
              "mostly_blank_count": 0,
              "mostly_blank_ms": 0,
              "pixels_blank": 0,
              "pixels_sampled": 0,
              "pixels_scrolled": 0,
              "sample_count": 0,
              "total_time_spent": 0,
            },
            "_mostlyBlankStartTime": null,
            "_samplesStartTime": null,
          }
        }
        horizontal={true}
        inversionStyle={null}
        item={
          Object {
            "id": 1,
          }
        }
        onUnmount={[Function]}
        onUpdateSeparators={[Function]}
        parentProps={
          Object {
            "CellRendererComponent": [Function],
            "activeSlideAlignment": "center",
            "activeSlideOffset": 20,
            "apparitionDelay": 0,
            "automaticallyAdjustContentInsets": false,
            "autoplay": false,
            "autoplayDelay": 1000,
            "autoplayInterval": 3000,
            "callbackOffsetMargin": 5,
            "containerCustomStyle": Object {},
            "contentContainerCustomStyle": Object {},
            "contentContainerStyle": Array [
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {},
            ],
            "data": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ],
            "decelerationRate": "fast",
            "directionalLockEnabled": true,
            "disableScrollViewPanResponder": false,
            "enableSnap": true,
            "firstItem": 0,
            "forwardedRef": [Function],
            "getItem": [Function],
            "getItemCount": [Function],
            "getItemLayout": [Function],
            "hasParallaxImages": false,
            "horizontal": true,
            "inactiveSlideOpacity": 0.7,
            "inactiveSlideScale": 0.9,
            "inactiveSlideShift": 0,
            "initialNumToRender": 6,
            "initialScrollIndex": 0,
            "inverted": false,
            "itemWidth": 1000,
            "keyExtractor": [Function],
            "layout": "default",
            "loop": false,
            "loopClonesPerSide": 3,
            "maxToRenderPerBatch": 10,
            "onLayout": [Function],
            "onMomentumScrollEnd": [Function],
            "onScroll": [Function],
            "onScrollIndexChanged": [Function],
            "onSnapToItem": [Function],
            "onTouchEnd": [Function],
            "onTouchStart": [Function],
            "overScrollMode": "never",
            "pinchGestureEnabled": false,
            "pointerEvents": "auto",
            "removeClippedSubviews": false,
            "renderItem": [Function],
            "scrollEnabled": true,
            "scrollEventThrottle": 1,
            "scrollsToTop": false,
            "shouldOptimizeUpdates": true,
            "showsHorizontalScrollIndicator": false,
            "showsVerticalScrollIndicator": false,
            "slideStyle": Object {},
            "sliderWidth": 1000,
            "snapToOffsets": Array [
              0,
              1000,
              2000,
            ],
            "style": Object {
              "flexDirection": "row",
              "width": 1000,
            },
            "useExperimentalSnap": false,
            "useScrollView": false,
            "viewabilityConfigCallbackPairs": Array [],
            "windowSize": 3,
          }
        }
        style={
          Array [
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ],
            Object {
              "zIndex": 3,
            },
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 1000,
            }
          }
        />
      </View>
      <View
        CellRendererComponent={[Function]}
        cellKey="flatlist-item-1"
        fillRateHelper={
          FillRateHelper {
            "_anyBlankStartTime": null,
            "_enabled": false,
            "_getFrameMetrics": [Function],
            "_info": Info {
              "any_blank_count": 0,
              "any_blank_ms": 0,
              "any_blank_speed_sum": 0,
              "mostly_blank_count": 0,
              "mostly_blank_ms": 0,
              "pixels_blank": 0,
              "pixels_sampled": 0,
              "pixels_scrolled": 0,
              "sample_count": 0,
              "total_time_spent": 0,
            },
            "_mostlyBlankStartTime": null,
            "_samplesStartTime": null,
          }
        }
        horizontal={true}
        inversionStyle={null}
        item={
          Object {
            "id": 2,
          }
        }
        onUnmount={[Function]}
        onUpdateSeparators={[Function]}
        parentProps={
          Object {
            "CellRendererComponent": [Function],
            "activeSlideAlignment": "center",
            "activeSlideOffset": 20,
            "apparitionDelay": 0,
            "automaticallyAdjustContentInsets": false,
            "autoplay": false,
            "autoplayDelay": 1000,
            "autoplayInterval": 3000,
            "callbackOffsetMargin": 5,
            "containerCustomStyle": Object {},
            "contentContainerCustomStyle": Object {},
            "contentContainerStyle": Array [
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {},
            ],
            "data": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ],
            "decelerationRate": "fast",
            "directionalLockEnabled": true,
            "disableScrollViewPanResponder": false,
            "enableSnap": true,
            "firstItem": 0,
            "forwardedRef": [Function],
            "getItem": [Function],
            "getItemCount": [Function],
            "getItemLayout": [Function],
            "hasParallaxImages": false,
            "horizontal": true,
            "inactiveSlideOpacity": 0.7,
            "inactiveSlideScale": 0.9,
            "inactiveSlideShift": 0,
            "initialNumToRender": 6,
            "initialScrollIndex": 0,
            "inverted": false,
            "itemWidth": 1000,
            "keyExtractor": [Function],
            "layout": "default",
            "loop": false,
            "loopClonesPerSide": 3,
            "maxToRenderPerBatch": 10,
            "onLayout": [Function],
            "onMomentumScrollEnd": [Function],
            "onScroll": [Function],
            "onScrollIndexChanged": [Function],
            "onSnapToItem": [Function],
            "onTouchEnd": [Function],
            "onTouchStart": [Function],
            "overScrollMode": "never",
            "pinchGestureEnabled": false,
            "pointerEvents": "auto",
            "removeClippedSubviews": false,
            "renderItem": [Function],
            "scrollEnabled": true,
            "scrollEventThrottle": 1,
            "scrollsToTop": false,
            "shouldOptimizeUpdates": true,
            "showsHorizontalScrollIndicator": false,
            "showsVerticalScrollIndicator": false,
            "slideStyle": Object {},
            "sliderWidth": 1000,
            "snapToOffsets": Array [
              0,
              1000,
              2000,
            ],
            "style": Object {
              "flexDirection": "row",
              "width": 1000,
            },
            "useExperimentalSnap": false,
            "useScrollView": false,
            "viewabilityConfigCallbackPairs": Array [],
            "windowSize": 3,
          }
        }
        prevCellKey="flatlist-item-0"
        style={
          Array [
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ],
            Object {
              "zIndex": 2,
            },
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 1000,
            }
          }
        />
      </View>
      <View
        CellRendererComponent={[Function]}
        cellKey="flatlist-item-2"
        fillRateHelper={
          FillRateHelper {
            "_anyBlankStartTime": null,
            "_enabled": false,
            "_getFrameMetrics": [Function],
            "_info": Info {
              "any_blank_count": 0,
              "any_blank_ms": 0,
              "any_blank_speed_sum": 0,
              "mostly_blank_count": 0,
              "mostly_blank_ms": 0,
              "pixels_blank": 0,
              "pixels_sampled": 0,
              "pixels_scrolled": 0,
              "sample_count": 0,
              "total_time_spent": 0,
            },
            "_mostlyBlankStartTime": null,
            "_samplesStartTime": null,
          }
        }
        horizontal={true}
        inversionStyle={null}
        item={
          Object {
            "id": 3,
          }
        }
        onUnmount={[Function]}
        onUpdateSeparators={[Function]}
        parentProps={
          Object {
            "CellRendererComponent": [Function],
            "activeSlideAlignment": "center",
            "activeSlideOffset": 20,
            "apparitionDelay": 0,
            "automaticallyAdjustContentInsets": false,
            "autoplay": false,
            "autoplayDelay": 1000,
            "autoplayInterval": 3000,
            "callbackOffsetMargin": 5,
            "containerCustomStyle": Object {},
            "contentContainerCustomStyle": Object {},
            "contentContainerStyle": Array [
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {},
            ],
            "data": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ],
            "decelerationRate": "fast",
            "directionalLockEnabled": true,
            "disableScrollViewPanResponder": false,
            "enableSnap": true,
            "firstItem": 0,
            "forwardedRef": [Function],
            "getItem": [Function],
            "getItemCount": [Function],
            "getItemLayout": [Function],
            "hasParallaxImages": false,
            "horizontal": true,
            "inactiveSlideOpacity": 0.7,
            "inactiveSlideScale": 0.9,
            "inactiveSlideShift": 0,
            "initialNumToRender": 6,
            "initialScrollIndex": 0,
            "inverted": false,
            "itemWidth": 1000,
            "keyExtractor": [Function],
            "layout": "default",
            "loop": false,
            "loopClonesPerSide": 3,
            "maxToRenderPerBatch": 10,
            "onLayout": [Function],
            "onMomentumScrollEnd": [Function],
            "onScroll": [Function],
            "onScrollIndexChanged": [Function],
            "onSnapToItem": [Function],
            "onTouchEnd": [Function],
            "onTouchStart": [Function],
            "overScrollMode": "never",
            "pinchGestureEnabled": false,
            "pointerEvents": "auto",
            "removeClippedSubviews": false,
            "renderItem": [Function],
            "scrollEnabled": true,
            "scrollEventThrottle": 1,
            "scrollsToTop": false,
            "shouldOptimizeUpdates": true,
            "showsHorizontalScrollIndicator": false,
            "showsVerticalScrollIndicator": false,
            "slideStyle": Object {},
            "sliderWidth": 1000,
            "snapToOffsets": Array [
              0,
              1000,
              2000,
            ],
            "style": Object {
              "flexDirection": "row",
              "width": 1000,
            },
            "useExperimentalSnap": false,
            "useScrollView": false,
            "viewabilityConfigCallbackPairs": Array [],
            "windowSize": 3,
          }
        }
        prevCellKey="flatlist-item-1"
        style={
          Array [
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ],
            Object {
              "zIndex": 1,
            },
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 1000,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    testID="pagination-testID"
  >
    <View
      accessible={false}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 20,
            "paddingVertical": 30,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 8,
            "paddingVertical": 0,
          },
        ]
      }
    >
      <View
        accessible={false}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 4,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#12A3BE",
              "borderRadius": 5,
              "height": 8,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
      <View
        accessible={false}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 4,
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "#D3D3D3",
              "borderRadius": 5,
              "height": 8,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
      <View
        accessible={false}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 4,
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "#D3D3D3",
              "borderRadius": 5,
              "height": 8,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
